### PR TITLE
fix: sorting hydration 

### PR DESCRIPTION
**Pull request by nsilva-spryker**
Closes [HRZ-2229](https://spryker.atlassian.net/browse/HRZ-2229)

### CODE REVIEW

File: libs/domain/search/sort/src/sort.component.ts [hunk 1]
```diff
@@ -29,6 +30,8 @@ export class SortComponent extends LitElement {
   protected renderSorting(
     sort: ProductListSort | undefined | null
```
Comment (galitskye) [inline libs/domain/search/sort/src/sort.component.ts]: My suggestion to make check before invoke `renderSorting` method. To avoid `undefined | null`

**Review by galitskye — CHANGES_REQUESTED** [review]

File: libs/domain/search/sort/src/sort.component.ts (hunk 1 above)
Comment (nsilva-spryker) [reply libs/domain/search/sort/src/sort.component.ts]: I keep it even more simple by using the sortingOptions directly in the method

File: libs/domain/search/sort/src/sort.component.ts [hunk 2]
```diff
@@ -24,13 +23,11 @@ export class SortComponent extends LitElement {
...
-    return this.renderSorting(this.sortingOptions);
+    return this.renderSorting();
   }
 
-  protected renderSorting(
-    sort: ProductListSort | undefined | null
-  ): TemplateResult {
-    const hasOptions = !!sort?.sortValues.length;
+  protected renderSorting(): TemplateResult {
+    const hasOptions = !!this.sortingOptions?.sortValues.length;
```
Comment (galitskye) [inline libs/domain/search/sort/src/sort.component.ts]: If I correct understand `sortValues` may be undefined or array. By trying take length from `undefined` will be an error.

Comment (nsilva-spryker) [reply libs/domain/search/sort/src/sort.component.ts]: The expression is checking whether sortingOptions is true and whether it has a non-zero sortValues array length.
If the sortValues array is empty or null, the expression returns 0. The current code could be sufficient

**Review by galitskye — APPROVED** [review]

**Review by dunqan — APPROVED** [review]

**Pull request merged**